torchgeo/datasets/ssl4eo_benchmark.py

```diff
@@ -0,0 +1,370 @@
+
+        Args:
+            index: index to return
+
+        Returns:
+            image and sample
+        """
+        img_path, mask_path = self.sample_collection[index]
+
+        sample = {
+            "image": self._load_image(img_path),
+            "mask": self._load_mask(mask_path),
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.sample_collection)
+
+    def retrieve_sample_collection(self) -> list[tuple[str, str]]:
+        """Retrieve paths to samples in data directory."""
+        img_paths = sorted(
+            glob.glob(
+                os.path.join(self.root, self.img_dir_name, "**", "all_bands.tif"),
+                recursive=True,
+            )
+        )
+        sample_collection: list[tuple[str, str]] = []
+        for img_path in img_paths:
+            mask_path = img_path.replace(self.img_dir_name, self.mask_dir_name).replace(
+                "all_bands.tif",
+                f"{self.mask_product}_{self.year_dict[self.input_sensor]}.tif",
+            )
+            sample_collection.append((img_path, mask_path))
+        return sample_collection
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load the input image.
+
+        Args:
+            path: path to input image
+
+        Returns:
+            image
+        """
+        with rasterio.open(path) as src:
+            image = src.read().astype(np.float32)
+        return torch.from_numpy(image)
+
+    def _load_mask(self, path: str) -> Tensor:
+        """Load the mask.
+
+        Args:
+            path: path to mask
+
+        Retuns:
+            mask
+        """
+        with rasterio.open(path) as src:
+            mask = src.read()
+
+        for k, v in self.ordinal_label_map[self.mask_product].items():
+            mask[mask == k] = v
+
+        return torch.from_numpy(mask).long()
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        ncols = 2
+        image = sample["image"][self.rgb_indices[self.input_sensor]].permute(1, 2, 0)
+        image = image.numpy() / 255
+
+        plt_cmap = ListedColormap(
+            np.stack(
+                [np.array(val) / 255 for val in self.cmaps[self.mask_product].values()],
+                axis=0,
+            )
+        )
```

Comment:
Forgot to say this during the CDL review, but don't we also need to map this to 134 classes before plotting? Should have asked you to add an example plot for CDL.

Reply:
updated CDL example at the top. Each available class has an assigned color already, I don't think you need to remap.

Reply:
I guess you would have to remap the ordinal values to then access the proper cmap values.